### PR TITLE
fix(authoring): fix saving of packages

### PR DESCRIPTION
**Pull request by petrjasek**
keep original etag/last_updated when setting linked_in_packages

### CODE REVIEW

File: server/apps/archive/archive_composite.py [hunk 1]
```diff
@@ -134,8 +134,11 @@ def update_link(self, package_id, assoc, delete=False):
         if not delete:
             two_way_links.append({'package': package_id})
 
+        # do update without changing etag/last_modified
         updates = self.get_item_update_data(item, two_way_links, delete)
-        get_resource_service(endpoint).patch(item_id, updates)
+        updates['_etag'] = item['_etag']
```
Comment (amagdas) [inline server/apps/archive/archive_composite.py]: Why is this needed? It looks weird to have two different docs having the same etag...

Comment (petrjasek) [inline server/apps/archive/archive_composite.py]: it's same doc. but you modify the etag every time you save a package. and if you edit this somewhere else it would get etag conflict

Comment (petrjasek) [inline server/apps/archive/archive_composite.py]: i mean what it does is setting the etag to previous value to avoid autogenering

Comment (amagdas) [inline server/apps/archive/archive_composite.py]: I understood that, but maybe a better solution would be to notify the client that the document updated so it can refetch the new document with the latest changes, including _etag...
My worry is that having an entity in mongo with the same _etag although they are not having the same content, it will break concurrency control...

Comment (petrjasek) [inline server/apps/archive/archive_composite.py]: it's single item, you update that same item, change there linked_in_packages, but keep the etag.
so if user (it might be other user, item can be locked etc) will save item it will still work. we can't reload that item for user who is editing content because someone else might pick it into a package imo

Comment (amagdas) [inline server/apps/archive/archive_composite.py]: I agree, I think a better solution is to use the Eve support for this: https://github.com/nicolaiarocci/eve/commit/d7e9bd024d9c1a5faad0e31b737d119cf3ee1fc7

Comment (petrjasek) [inline server/apps/archive/archive_composite.py]: thing is this is after eve handling, updating a related item, not the package. it's making changes directly

Comment (ioanpocol) [inline server/apps/archive/archive_composite.py]: I think that a good approach is that the saved etag to have the same value as the one calculated on values, otherwise we will have some issues like those that we have now for user preferences that are failing on save because of this matter.

Comment (petrjasek) [inline server/apps/archive/archive_composite.py]: here the point is not to change it when we change values

**Comment by ioanpocol** [comment]
:+1:

File: server/features/steps/steps.py [hunk 2]
```diff
@@ -289,6 +289,21 @@ def step_impl_when_post_url(context, url):
...
 
 
+def get_response_etag(response):
+    return json.loads(response.get_data())['_etag']
+
+
+@when('we save etag')
+def step_when_we_save_etag(context):
+    context.etag = get_response_etag(context.response)
+    print(context.etag)
```
Comment (mugurrus) [inline server/features/steps/steps.py]: delete print

**Comment by mugurrus** [comment]
:+1:
